app/scripts/lib/snap-keyring/snap-keyring.ts

```diff
@@ -145,118 +184,117 @@ export const snapKeyringBuilder = (
         // If snap is preinstalled and does not request confirmation, skip the confirmation dialog
         const skipConfirmation =
           isSnapPreinstalled(snapId) && !displayConfirmation;
-        // If confirmation dialog is skipped, we consider the account creation to be confirmed
-        let confirmationResult = skipConfirmation;
-        let confirmationApprovalId = '';
-        try {
-          if (!skipConfirmation) {
-            const { id } = controllerMessenger.call(
-              'ApprovalController:startFlow',
-            );
-            confirmationApprovalId = id;
-            confirmationResult = Boolean(
-              await controllerMessenger.call(
-                'ApprovalController:addRequest',
-                {
-                  origin: snapId,
-                  type: SNAP_MANAGE_ACCOUNTS_CONFIRMATION_TYPES.confirmAccountCreation,
-                },
-                true,
-              ),
-            );
-          }
+        // If confirmation dialog are skipped, we consider the account creation to be confirmed until the account name dialog
+        let confirmationResult =
+          skipConfirmation ||
+          (await showAccountCreationDialog(snapId, controllerMessenger));
 
-          if (confirmationResult) {
-            try {
-              await handleUserInput(confirmationResult);
-              await persistKeyringHelper();
-              setSelectedAccountHelper(address);
-              const internalAccount = controllerMessenger.call(
-                'AccountsController:getAccountByAddress',
+        // We need to temporarily add the account to the keyring to show the account name suggestion dialog
+        await handleUserInput(confirmationResult);
+        await persistKeyringHelper();
```

Comment:
This used to be checking for `if (confirmationResult)` before persisting anything.

If the user denied, we were still calling `handleUserInput` but we were also raising an exception (without persisting).

Reply:
Yes I realized that now! I changed the flow to now have an `initialConfirmationResult` and that will throw an error and end the flow if is returns positive, otherwise, it will proceed with the naming modal.
Commit with fix [352d9ae](https://github.com/MetaMask/metamask-extension/pull/25191/commits/352d9ae33c55692a2305c10da3a0fcb3757f911a)